packages/@sanity/core/src/actions/dataset/listDatasetCopyJobs.js

```diff
@@ -0,0 +1,112 @@
+const {Table} = require('console-table-printer')
+
+module.exports = async function listDatasetCopyJobs(args, context) {
+  const {apiClient, output, chalk} = context
+  const flags = args.extOptions
+  const client = apiClient()
+  const projectId = client.config().projectId
+  const query = {}
+  let response
+
+  if (flags.offset && flags.offset >= 0) {
+    query.offset = flags.offset
+  }
+  if (flags.limit && flags.limit > 0) {
+    query.limit = flags.limit
+  }
+
+  try {
+    response = await client.request({
+      method: 'GET',
+      uri: `/projects/${projectId}/datasets/copy`,
+      query,
+    })
+  } catch (error) {
+    if (error.statusCode) {
+      output.print(`${chalk.red(`Dataset copy list failed:\n${error.response.body.message}`)}\n`)
+    } else {
+      output.print(`${chalk.red(`Dataset copy list failed:\n${error.message}`)}\n`)
+    }
+  }
+
+  if (response && response.length > 0) {
+    const p = new Table({
+      title: 'Dataset copy jobs for this project in descending order',
+      columns: [
+        {name: 'id', title: 'Job ID', alignment: 'left'},
+        {name: 'sourceDataset', title: 'Source Dataset', alignment: 'left'},
+        {name: 'targetDataset', title: 'Target Dataset', alignment: 'left'},
+        {name: 'state', title: 'State', alignment: 'left'},
+        {name: 'withHistory', title: 'With history', alignment: 'left'},
+        {name: 'timeStarted', title: 'Time started', alignment: 'left'},
+        {name: 'timeTaken', title: 'Time taken', alignment: 'left'},
+      ],
+    })
+
+    response.forEach((job) => {
+      const {id, state, createdAt, updatedAt, sourceDataset, targetDataset, withHistory} = job
+
+      const createdAtInSeconds = new Date(createdAt).getTime() / 1000
+      const nowInSeconds = new Date().getTime() / 1000
+      const timePassedInSeconds = nowInSeconds - createdAtInSeconds
+      const timeStarted = convertTimestampToDaysHoursMinsSecs(timePassedInSeconds)
+
+      const updatedAtInSeconds = new Date(updatedAt).getTime() / 1000
+      const timeTakenInSeconds = updatedAtInSeconds - createdAtInSeconds
+      const timeTaken = convertTimestampToDaysHoursMinsSecs(timeTakenInSeconds)
+
+      let color
+      switch (state) {
+        case 'completed':
+          color = 'green'
+          break
+        case 'failed':
+          color = 'red'
+          break
+        case 'pending':
+          color = 'yellow'
+          break
+        default:
+          color = ''
+      }
+
+      p.addRow(
+        {
+          id,
+          state,
+          withHistory,
+          timeStarted: `${timeStarted} ago`,
+          timeTaken,
+          sourceDataset,
+          targetDataset,
+        },
+        {color}
+      )
+    })
+
+    p.printTable()
+  } else {
+    output.print("This project doesn't have any dataset copy jobs")
+  }
+}
+
+function convertTimestampToDaysHoursMinsSecs(timePassedInSeconds) {
```

Comment:
Can I suggest using shorthand formatting for these, to compress the table some more? Eg instead of `3 days 22 hours 32 minutes 6 seconds` it would be `3d 22h 32m 6s`